src/v/oncore.h

```diff
@@ -10,6 +10,9 @@
  */
 #pragma once
 #include "utils/source_location.h"
+#include "vassert.h"
+
+#include <seastar/core/smp.hh>
```

Comment:
Nice. I guess this is essentially included everywhere, so hiding it in the definition file wasn't useful.

Reply:
yeh